test/app-tap/uri.test.lua

```diff
@@ -96,6 +96,76 @@ local function test_parse(test)
 end
 
+local function test_parse_ipv6_format_by_rfc2732(test)
+    -- The literal format of an IPv6 address can contain the following
+    -- characters [0-9;a-f;A-F]. It was discovered in that the
+    -- uri does not support the characters A-F, although it should
+    -- according to the RFC 2732.
+    test:plan(36)
+
+    local u
+
+    u = uri.parse("[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]")
+    test:is(u.host, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'host')
+    test:is(u.ipv6, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'ipv6')
+
+    u = uri.parse("[1080:0:0:0:8:800:200C:4171]")
+    test:is(u.host, '1080:0:0:0:8:800:200C:4171', 'host')
+    test:is(u.ipv6, '1080:0:0:0:8:800:200C:4171', 'ipv6')
+
+    u = uri.parse("[3ffe:2a00:100:7031::1]")
+    test:is(u.host, '3ffe:2a00:100:7031::1', 'host')
+    test:is(u.ipv6, '3ffe:2a00:100:7031::1', 'ipv6')
+
+    u = uri.parse("[1080::8:800:200C:417A]")
+    test:is(u.host, '1080::8:800:200C:417A', 'host')
+    test:is(u.ipv6, '1080::8:800:200C:417A', 'ipv6')
+
+    u = uri.parse("[::FFFF:129.144.52.38]")
+    test:is(u.host, '::FFFF:129.144.52.38', 'host')
+    test:is(u.ipv6, '::FFFF:129.144.52.38', 'ipv6')
+
+    u = uri.parse("[2010:836b:4179::836b:4179]")
+    test:is(u.host, '2010:836b:4179::836b:4179', 'host')
+    test:is(u.ipv6, '2010:836b:4179::836b:4179', 'ipv6')
+
+    u = uri.parse("http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80/index.html")
+    test:is(u.host, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'ipv6')
+    test:is(u.service, '80', 'service')
+    test:is(u.path, '/index.html', 'path')
+
+    u = uri.parse("http://[1080:0:0:0:8:800:200C:417A]/index.html")
+    test:is(u.host, '1080:0:0:0:8:800:200C:417A', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, '1080:0:0:0:8:800:200C:417A', 'ipv6')
+    test:is(u.path, '/index.html', 'path')
+
+    u = uri.parse("http://[3ffe:2a00:100:7031::1]")
+    test:is(u.host, '3ffe:2a00:100:7031::1', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, '3ffe:2a00:100:7031::1', 'ipv6')
+
+    u = uri.parse("http://[1080::8:800:200C:417A]/foo")
+    test:is(u.host, '1080::8:800:200C:417A', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, '1080::8:800:200C:417A', 'ipv6')
+    test:is(u.path, '/foo', 'path')
+
+    u = uri.parse("http://[::FFFF:129.144.52.38]:80/index.html")
+    test:is(u.host, '::FFFF:129.144.52.38', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, '::FFFF:129.144.52.38', 'ipv6')
+    test:is(u.service, '80', 'service')
+    test:is(u.path, '/index.html', 'path')
+
+    u = uri.parse("http://[2010:836B:4179::836B:4179]")
+    test:is(u.host, '2010:836B:4179::836B:4179', 'host')
+    test:is(u.scheme, 'http', 'scheme')
+    test:is(u.ipv6, '2010:836B:4179::836B:4179', 'ipv6')
```

Comment:
Please use parameterization for testcases. With luatest [^1] it will be easier. Do it in TAP or in luatest is up to you.

[^1]: https://github.com/tarantool/luatest/?tab=readme-ov-file#parametrization

Reply:
I moved the tests to a separate file and did the parameterization

Reply:
Thanks!